packages/devtools_app/lib/src/shared/memory/retainers.dart

```diff
@@ -4,23 +4,41 @@
 
 import 'dart:typed_data';
 
-typedef IsRetainer = bool Function(int index);
+/// Returns true if the given object can retain other objects from garbage collection.
+typedef IsWeak = bool Function(int index);
+
+/// List of references for the given object.
 typedef References = List<int> Function(int index);
+
+/// Shallow size of the given object.
 typedef ShallowSize = int Function(int index);
 
 typedef ShortestRetainersResult = ({
+  /// Retainer for each object in the graph.
+  ///
+  /// 0 means no-retainer.
```

Comment:
Should this be "a list with a single element zero [0] implies there is no retainer for the object."?